Dependabot started to fail with `reject': undefined method `blank?' for ">= 2.1.0, < 2.1.29":String (NoMethodError)

No update needed for MicroElements.Swashbuckle.FluentValidation 5.7.0
Checking if Microsoft.AspNetCore.Authentication.JwtBearer 7.0.3 needs updating
bin/update-script.rb:314:in `reject': undefined method `blank?' for ">= 2.1.0, < 2.1.29":String (NoMethodError)

    vulnerable_versions = vulnerable_versions.reject(&:blank?).reject(&:empty?)
                                             ^^^^^^^
	from bin/update-script.rb:314:in `block in security_advisories_for'
	from bin/update-script.rb:303:in `map'
	from bin/update-script.rb:303:in `security_advisories_for'
	from bin/update-script.rb:546:in `block in <main>'
	from bin/update-script.rb:533:in `each'
	from bin/update-script.rb:533:in `<main>'
##[error]The process '/usr/bin/docker' failed with exit code 1

--- a/updater/bin/update-script.rb
+++ b/updater/bin/update-script.rb
@@ -305,14 +305,14 @@ def security_advisories_for(dep)
     safe_versions = (adv["patched-versions"] || []) +
                     (adv["unaffected-versions"] || [])
 
-    # Filter out nil (blank objects) and empty strings which is necessary for situations
+    # Filter out nil (using .compact), white spaces and empty strings which is necessary for situations
     # where the API response contains null that is converted to nil, or it is an empty
     # string. For example, npm package named faker does not have patched version as of 2023-01-16
     # See: https://github.com/advisories/GHSA-5w9c-rv96-fr7g for npm package
     # This ideally fixes
     # https://github.com/tinglesoftware/dependabot-azure-devops/issues/453#issuecomment-1383587644
-    vulnerable_versions = vulnerable_versions.reject(&:blank?).reject(&:empty?)
-    safe_versions = safe_versions.reject(&:blank?).reject(&:empty?)
+    vulnerable_versions = vulnerable_versions.compact.reject { |v| v.strip.empty? }
+    safe_versions = safe_versions.compact.reject { |v| v.strip.empty? }
     next if vulnerable_versions.empty? && safe_versions.empty?
 
     Dependabot::SecurityAdvisory.new(

